### ISSUE TITLE
Flaky-test: BasicEndToEndTest.testPatternMultiTopicsConsumerPubSub

### ISSUE THREAD

**Issue by tisonkun** [apache/pulsar#17008]
### Search before asking

- [X] I searched in the [issues](https://github.com/apache/pulsar/issues) and found nothing similar.

### Example failure

https://github.com/apache/pulsar/runs/7739162912?check_suite_focus=true#step:11:185

### Exception stacktrace

```
/pulsar/pulsar-client-cpp/tests/BasicEndToEndTest.cc:2184: Failure
Expected equality of these values:
  ResultTimeout
    Which is: TimeOut
  consumer.receive(m, 1000)
    Which is: Ok
[  FAILED  ] BasicEndToEndTest.testPatternMultiTopicsConsumerPubSub (1173 ms)
[----------] 1 test from BasicEndToEndTest (1173 ms total)

[----------] Global test environment tear-down
[==========] 1 test from 1 test suite ran. (1173 ms total)
[  PASSED  ] 0 tests.
[  FAILED  ] 1 test, listed below:
[  FAILED  ] BasicEndToEndTest.testPatternMultiTopicsConsumerPubSub
```

This can be relevant to https://github.com/apache/pulsar/issues/16802.

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

**Comment by tisonkun** [comment]
It seems in that attempt multiple tasks failed due to timeout. Perhaps it's an issue with GitHub CI resource unstable.

Keep it open for a few days to see if there're other instances. If no further reported, anyone can close it as invalid.

**Comment by mattisonchao** [comment]
I think this problem may caused by #16968 and fixed by #17018.
Currently, the fix PR already merged, maybe we can close this issue until we got it again.

**Comment by tisonkun** [comment]
@mattisonchao OK.
